source/platform_apple.cmake

```diff
@@ -0,0 +1,47 @@
+if( NOT APPLE)
+    message(FATAL_ERROR "Don't include this on non-macOS platforms.")
+endif()
+
+include(CheckIncludeFileCXX)
+check_include_file_cxx(filesystem HAVE_FILESYSTEM)
+
+if(NOT HAVE_FILESYSTEM)
+  message(FATAL_ERROR "filesystem header NOT found. Please use LLVM Clang or upgrade to Xcode 11.")
+endif()
+
+# Darwin 19 (macOS 10.15) adds the std::filesystem::path.
+if(CMAKE_SYSTEM_VERSION VERSION_LESS 19)
+    # This fixes the "path explicitly marked unavailable here" error.
+    add_definitions("-D_LIBCPP_DISABLE_AVAILABILITY")
+    # Final location for clang dylibs and aliases.
+    set(INSTALL_PATH_LIBCPPDYLIB "${ROOT_INSTALL_PREFIX}/libexec")
+    get_filename_component(CLANG_INSTALL_PREFIX ${CMAKE_CXX_COMPILER}/../../ ABSOLUTE)
+    execute_process(
+        COMMAND ${CMAKE_COMMAND} -E copy
+            ${CLANG_INSTALL_PREFIX}/lib/libc++.1.0.dylib
+            ${CLANG_INSTALL_PREFIX}/lib/libc++.1.dylib
+            ${CLANG_INSTALL_PREFIX}/lib/libc++.dylib
+
+            "${INSTALL_PATH_LIBCPPDYLIB}"
+        ERROR_VARIABLE ERROR_COPY_LIB
+    )
+    if(ERROR_COPY_LIB)
+        message(WARNING "${ERROR_COPY_LIB}")
+    endif()
+    # https://libcxx.llvm.org/docs/UsingLibcxx.html#alternate-libcxx
+    add_compile_options("-stdlib=libc++;-nostdinc++;-I${CLANG_INSTALL_PREFIX}/include/c++/v1")
+    # For linking and testing, executables should find the dylibs here.
+    add_link_options("-L${CLANG_INSTALL_PREFIX}/lib;-Wl,-rpath,${INSTALL_PATH_LIBCPPDYLIB}")
+    # For distribution and running from installed directories, executables should find the dylibs here.
+    add_link_options("-Wl,-rpath,@executable_path/../libexec")
+endif()
+
+macro(remove_temporary_rpath
```

Comment:
Unused macro right now, it needs all executable names to remove the info of the build maker.
run otool -l path/to/executable
It is not necessary, just cosmetic.

Reply:
Can you remove it now, and introduce it in a separate PR maybe when it's complete?

Reply:
It is complete, I guess, haven't run it. If there's an easily findable list of all executables, it should work. It just needs your demand to remove your build path from the binary.